daemon/api.go

```diff
@@ -1864,6 +1864,11 @@ func changeInterfaces(c *Command, r *http.Request, user *auth.UserState) Respons
 	}
 
 	change := newChange(st, a.Action+"-snap", summary, tasksets, affected)
+	// Special case for disconnect: if no connections exist, we just report Done.
+	// Note that for every other case ResolveDisconnect will give an error.
+	if len(tasksets) == 0 {
```

Comment:
As discussed IRL, I think this should behave exactly like "snap connect foo:bar froz:bot" when those things are already connected, that is do nothing (no tasks) successfully.